test/unit/buffer_spec.lua

```diff
@@ -212,92 +213,164 @@ describe('buffer functions', function()
 
   describe('build_stl_str_hl', function()
 
-    local output_buffer = to_cstr(string.rep(" ", 100))
-
-    local build_stl_str_hl = function(pat)
-      return buffer.build_stl_str_hl(globals.curwin,
-                                     output_buffer,
-                                     100,
-                                     to_cstr(pat),
-                                     false,
-                                     32,
-                                     80,
-                                     NULL,
-                                     NULL)
-    end
-
-    it('should copy plain text', function()
-      local width = build_stl_str_hl("this is a test")
-
-      eq(14, width)
-      eq("this is a test", helpers.ffi.string(output_buffer, width))
-
+    before_each(function()
+      buffer_width = 100
+      output_buffer = to_cstr(string.rep(" ", buffer_width))
+      -- Currently assumes fillchar = ' '
+
+      build_stl_str_hl = function (arg)
+        if not arg.fillchar then
+          arg.fillchar = 0x20
+        end
+
+        return buffer.build_stl_str_hl(globals.curwin,
+                                       output_buffer,
+                                       buffer_width,
```

Comment:
This should not be named “width”, it is length of the buffer in bytes. Width is for screen cells.